pkgs/development/compilers/rust/rustc.nix

```diff
@@ -62,6 +63,13 @@ stdenv.mkDerivation {
                 ++ optional (targets != []) "--target=${target}"
                 ++ optional (!forceBundledLLVM) "--llvm-root=${llvmShared}";
 
+  # The boostrap.py will generated a Makefile that then executes the build.
+  # The BOOTSTRAP_ARGS used by this Makefile must include all flags to pass
+  # to the bootstrap builder.
+  postConfigure = ''
+    substituteInPlace Makefile --replace 'BOOTSTRAP_ARGS :=' 'BOOTSTRAP_ARGS := --jobs $(NIX_BUILD_CORES)'
```

Comment:
Is this for the llvm build?

Reply:
Not familiar with the rust build chain, but AFAICT:

This will impact the number of cores used by the LLVM build. However, this change targets the args passed to the rustc bootstrap builder. Which has a "jobs" arg:

* https://github.com/rust-lang/rust/blob/master/src/bootstrap/flags.rs#L122

The docs do not explicitly state "only LLVM" but that seems to be the only build actually impacted. *shrug*

Specifically this modifies

* https://github.com/rust-lang/rust/blob/master/src/bootstrap/mk/Makefile.in

Which is copied by the configure script to form the top level Makefile.

Though, now that I look at flags.rs closer: Maybe a "JOBS" env variable would work as well?

Reply:
I was asking about llvm specifically because I noticed that it does indeed use more resources than it's supposed to.

Reply:
Understood.

I did try this change using a simple `JOBS` env variable. No such luck. Patching `Makefile.in` is the only version I can get to work as expected.